Duplicati/Library/Main/Operation/Backup/BackendUploader.cs

```diff
@@ -126,75 +129,51 @@ public Task Run()
+                            uploadsInProgress = 0;
+                            m_operationUpdater.UpdatePhase(OperationPhase.Paused);
+                            m_progressUpdater.Pause();
+                            if (!await m_taskReader.TransferProgressAsync.ConfigureAwait(false))
+                                break;
+                            m_progressUpdater.Resume();
                         }
+                        else if (!await m_taskReader.TransferProgressAsync.ConfigureAwait(false))
+                            break;
 
-                        if (req is VolumeUploadRequest volumeUpload)
-                        {
-                            if (volumeUpload.IndexVolume == null)
-                                worker.Task = Task.Run(() => UploadFileAsync(volumeUpload.BlockEntry, worker, m_cancelTokenSource.Token));
-                            else
-                                worker.Task = Task.Run(() => UploadBlockAndIndexAsync(volumeUpload, worker, m_cancelTokenSource.Token));
+                        if (request == null)
+                            continue;
 
-                            lastSize = volumeUpload.BlockVolume.SourceSize;
-                            uploadsInProgress++;
-                        }
-                        else if (req is FilesetUploadRequest filesetUpload)
-                        {
-                            worker.Task = Task.Run(() => UploadVolumeWriter(filesetUpload.Fileset, worker, m_cancelTokenSource.Token));
-                            uploadsInProgress++;
-                        }
-                        else if (req is IndexVolumeUploadRequest indexUpload)
-                        {
-                            worker.Task = Task.Run(() => UploadVolumeWriter(indexUpload.IndexVolume, worker, m_cancelTokenSource.Token));
-                            uploadsInProgress++;
-                        }
-                        else if (req is FlushRequest flush)
+                        await HandleRequest(request).ConfigureAwait(false);
+                        uploadsInProgress++;
+                        if (request is FlushRequest)
                         {
-                            while (workers.Any())
-                            {
-                                Task finishedTask = await Task.WhenAny(workers.Select(w => w.Task)).ConfigureAwait(false);
-                                if (finishedTask.IsFaulted)
-                                {
-                                    flush.TrySetCanceled();
-                                    ExceptionDispatchInfo.Capture(finishedTask.Exception).Throw();
-                                }
-
-                                workers.RemoveAll(w => w.Task == finishedTask);
-                            }
-                            
-                            flush.SetFlushed(lastSize);
                             uploadsInProgress = 0;

```

Comment:
Is this assignment necessary?  The `uploadsInProgress` variable looks to be unused in this code path.